### DISCUSSION TITLE
Why `lib` is generated by babel with loose method?

### DISCUSSION THREAD

**Discussion by rpokrovskij** [floating-ui/floating-ui#1374] [Q&A] [answered]
Why `lib` is generated by babel with loose method? I would expect that since this is not a bundle that should work inside browser it could be left in original ES8/ES7. What I miss? Is it very complex for "end user" to transpile it later (during his app bundling)?

**Comment by FezVrasta** [comment, marked as answer]
The Babel configuration it's shared across the different builds. We could change the loose mode for the lib target.

**Reply by rpokrovskij** [reply]
It seems like there is no easy way to transpile Typescript references before they get processed by TypeScript.

https://stackoverflow.com/questions/50090972/typescript-transpile-es6-js-dependency-to-es5 

From popper perspective there is no advantages to complicate its users live with new plugins or several lib folders.  

P.S. I study your building process since it has its own high value as independent solution... Excuse me my naive questions.

**Reply by FezVrasta** [reply]
The app is built with Flow, not Typescript, so that shouldn't be an issue

**Reply by rpokrovskij** [reply]
I think this is an issue for  "end users" who develop theirs web pages with typescript, reference lib folder, and bundle with ts-loader. They can't simply "transpile on the fly" (no typescript instruments for it).  Does it make sense?
On contrast  "end users" who uses babel can transpile dependencies (manipulating exclude).

P.S. of course there is always "transpiled" esm bundle version for ts users (tree shaking should work over esm modules)...
